### ISSUE TITLE
Infinite loop when using take('*')

### ISSUE THREAD

**Issue by aikoven** [yelouafi/redux-saga#61]
Yielding effect `take('*')` results in dispatching monitor action which can be again consumed by the same effect. Simplest way to reproduce:

```js
function* saga() {
  while (true) {
    yield take('*');
  }
}

**Comment by yelouafi** [comment]
Yes. The middleware should not redispatch monitor events to sagas

**Comment by aft-luke** [comment]
Looks like this was taken care of https://github.com/yelouafi/redux-saga/blob/no-promise/src/middleware.js#L25-L27

**Comment by yelouafi** [comment]
Should be fixed by 0.6.0

**Comment by aikoven** [comment]
Great, thanks!
